### PR TITLE
[GEOS-9403] MongoDB Plugin breaks GeoServer Admin UI

### PR DESCRIPTION
**Pull request by fernandor777**
This PR fixes an Spring initialization methods scan issue with optional AppSchema dependencies classes.

JIRA issue:
https://osgeo-org.atlassian.net/browse/GEOS-9403

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.geotools.data.complex.FeatureTypeMapping;
+import org.geotools.data.mongodb.MongoDataStore;
+import org.opengis.feature.type.Name;
+import org.springframework.http.HttpStatus;
+
+/** Util methods for AppSchema centric logic. */
+class AppSchemaUtils {
+
+    /**
+     * Returns the internal MonfoDB store from the AppSchema data store with the provided store ID.
```
Comment (nmco) [inline src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java:25]: MonfoDB -> MongoDB

Comment (nmco) [reply src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java:25]: AppSchema -> App-Schema

File: src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            throws IOException {
+        for (Name etn : appSchemaStore.getNames()) {
+            FeatureTypeMapping featureTypeMapping = appSchemaStore.getMappingByName(etn);
+            if (Objects.equals(mongoStore, featureTypeMapping.getSource().getDataStore()))
+                return featureTypeMapping.getSourceDatastoreId().orElse(null);
+        }
+        return null;
+    }
+
+    static void fillMongoStoresSet(
```
Comment (nmco) [inline src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java:76]: JavaDoc?

**Comment by nmco** [comment]
Can you explain a little bit the fix you did to help me udnerstand, I got a bit lost :smile: to me it looks like you reorganized the class whit utils methods.

**Comment by fernandor777** [comment]
@nmco of course, the facts are:
- Spring context initializer scans for the components methods definitions and its involved types on startup phase.
- When Spring scans the ```MongoStoreRestController``` class, it scans the private methods as well and finds the App-Schema related classes and starts to try to get its definition, causing a ClassNotFoundException breaking the Spring initializing stage and the components startup.
- Moving the private App-Schema utility methods to another class avoids this component class private methods scan failure.

File: src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java (hunk 1 above)
Comment (fernandor777) [reply src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java:25]: Fixed, thanks

File: src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java (hunk 2 above)
Comment (fernandor777) [reply src/extension/mongodb/src/main/java/org/geoserver/rest/service/AppSchemaUtils.java:76]: Added

**Comment by aaime** [comment]
@nmco satisfied with the explaination? is there anything else that needs fixing here?

**Pull request merged**